python/lsst/ctrl/mpexec/cli/opt/options.py

```diff
@@ -440,6 +445,46 @@
     help=unwrap("""Names of input dataset types that should not be mocked."""),
 )
 
+
+def parse_mock_failure(
+    ctx: click.Context, param: click.Option, value: Iterable[str] | None
+) -> Mapping[str, tuple[str, type[Exception] | None]]:
+    """Parse the --mock-failure option values into the mapping accepted by
+    `lsst.pipe.base.tests.mocks.mock_task_defs`.
+    """
+    result: dict[str, tuple[str, type[Exception] | None]] = {}
+    if value is None:
+        return result
+    for entry in value:
+        try:
+            task_label, error_type_name, where = entry.split(":")
+        except ValueError:
+            raise click.UsageError(
+                f"Invalid value for --mock-failure option: {entry!r}; "
+                "expected a string of the form 'task:error:where'."
+            ) from None
+        error_type = doImportType(error_type_name) if error_type_name else None
+        result[task_label] = (where, error_type)
+    return result
+
+
+mock_failure_option = MWOptionDecorator(
+    "--mock-failure",
+    callback=parse_mock_failure,
+    metavar="LABEL:EXCEPTION:WHERE",
+    default=None,
+    multiple=True,
+    help=unwrap(
+        """Specifications for tasks that should be configured to fail
+        when mocking execution.  This is a colon-separated 3-tuple,
+        where the first entry the task label, the second the
+        fully-qualified exception type (empty for ValueError, and the
+        third a string (usually quoted) of the form passed to --where,
```

Comment:
"usually quoted" may be a bit confusing. I guess quoting is needed for shell only, not for the Python code that parses it?

Reply:
Yes, just for the shell.  Reworded to that effect.